### PR TITLE
In this commit I have added the testing and implemented for patch-art…

### PR DESCRIPTION
**Pull request by rachelkwok00**
…icles

### CODE REVIEW

File: models/model.js [hunk 1]
```diff
@@ -84,4 +84,23 @@ function addComment(article_id, newComment) {
 
 }
 
-module.exports = { selectTopics, getFile, selectArticleById, selectAllArticles, selectArticleComment, addComment }
\ No newline at end of file
+function changeVotes(article_id, newComment) {
+  console.log('in model')
```
Comment (B7Bprog) [inline models/model.js]: Please remove all debugging logs.

File: controllers/controller.js [hunk 2]
```diff
@@ -90,4 +90,23 @@ const {
       .catch(err => {
        next(err);
       })
-  };
\ No newline at end of file
+  };
+
+  exports.patchArticle = (req, res, next) => {
+    console.log('in controller')
```
Comment (B7Bprog) [inline controllers/controller.js]: Please remove debugging logs.

File: controllers/controller.js [hunk 3]
```diff
@@ -90,4 +90,23 @@ const {
...
+    const { article_id } = req.params;
+    const voteIncrement  = req.body;
+      
+    Promise.all([
```
Comment (B7Bprog) [inline controllers/controller.js]: You are using 'RETURNING' in your query, so you should get the patched object back even without selecting the article.

File: __tests__/endpoint.test.js [hunk 4]
```diff
@@ -332,3 +332,72 @@ describe('POST/api/articles/:article_id/comment', () => {
...
+      const testVote = { vote_increment : 6 }
+
+    return request(app)
+      .patch("/api/articles/1").send(testVote)
+      .expect(200)
+      .then((response) => {
+
+        const newVote = oldVote + testVote.vote_increment
+
+        const article= response.body.article
```
Comment (B7Bprog) [inline __tests__/endpoint.test.js]: Article should be an object instead of an array.

File: __tests__/endpoint.test.js [hunk 5]
```diff
@@ -332,3 +332,72 @@ describe('POST/api/articles/:article_id/comment', () => {
...
+         
+        expect(article.length).toBe(2)
```
Comment (B7Bprog) [inline __tests__/endpoint.test.js]: Why is the length 2 here? We should get a single updated article back. This should be an object (not an array).

File: __tests__/endpoint.test.js [hunk 6]
```diff
@@ -332,3 +332,72 @@ describe('POST/api/articles/:article_id/comment', () => {
...
+          expect(obj).toHaveProperty("body", expect.any(String))
+          expect(obj).toHaveProperty("votes", expect.any(Number))
+          
+          
+        })
+       
+      });
+    })
+  });  
+  test('PATCH:400 sends a error when invalid is passed', () => {
```
Comment (B7Bprog) [inline __tests__/endpoint.test.js]: Typo in test description: 'invalid is' should probably be 'invalid id'.

File: __tests__/endpoint.test.js [hunk 7]
```diff
@@ -332,3 +332,72 @@ describe('POST/api/articles/:article_id/comment', () => {
...
+
+    return db.query('SELECT * FROM articles WHERE article_id = 1').then ((result)=>{
```
Comment (B7Bprog) [inline __tests__/endpoint.test.js]: Why do you need this SQL query in this test?

File: __tests__/endpoint.test.js [hunk 8]
```diff
@@ -332,3 +332,72 @@ describe('POST/api/articles/:article_id/comment', () => {
...
+          expect(obj).toHaveProperty("article_img_url", expect.any(String))
+          expect(obj).toHaveProperty("body", expect.any(String))
+          expect(obj).toHaveProperty("votes", expect.any(Number))
+          
+          
+        })
+       
+      });
+    })
+  });
```
Comment (B7Bprog) [inline __tests__/endpoint.test.js:361]: Please add another test (400) where the 'vote_increment' property is not a number.

File: __tests__/endpoint.test.js [hunk 9]
```diff
@@ -332,3 +332,72 @@ describe('POST/api/articles/:article_id/comment', () => {
   });
 
 })
+
+describe('PATCH/api/articles/:article_id', () => {
+  test('PATCH:200 sends a single article object to the client with vote increment', () => {
+
+    return db.query('SELECT * FROM articles WHERE article_id = 1').then ((result)=>{
```
Comment (B7Bprog) [inline __tests__/endpoint.test.js]: Instead of querying the database here, we could just give the actual number of votes.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Rachel, please see my comments, refactor your code / add test, then please send this PR back to us with the updates.

File: __tests__/endpoint.test.js [hunk 10]
```diff
@@ -332,3 +332,85 @@ describe('POST/api/articles/:article_id/comment', () => {
...
+
+    return request(app)
+      .patch("/api/articles/1").send(testVote)
+      .expect(200)
+      .then((response) => {
+
+        const articleObj= response.body
+      
+        expect(articleObj).toBeInstanceOf(Object)
+
```
Comment (Augs0) [inline __tests__/endpoint.test.js]: If we are patching an article, we are patching only one, so we don't actually need to iterate over articles here

File: controllers/controller.js (hunk 3 above)
Comment (Augs0) [reply controllers/controller.js]: As Béla said above, we don't need to select the article. Simply using your changeVotes function will be enough. If it returns nothing, the article does not exist.

File: __tests__/endpoint.test.js [hunk 11]
```diff
@@ -332,3 +332,85 @@ describe('POST/api/articles/:article_id/comment', () => {
...
+      .expect(400)
+      .then((response) => {
+
+        expect(response.body.msg).toBe('Bad request');
+        
+      });
+    })
+    test('PATCH:400 sends a error when object key is not vote_increment is incorrect', () => {
+
+      const testVote = { 789 : 9 }
```
Comment (Augs0) [inline __tests__/endpoint.test.js]: I believe we cannot use numbers as column_names in PSQL, so I would consider changing 789 here to a string to make sure we are testing having the wrong key and not the wrong data type

**Review by Augs0 — CHANGES_REQUESTED** [review]
This is looking really good, well done! There are a few things for you to go back and edit slightly but this is almost there ✨

File: __tests__/endpoint.test.js [hunk 12]
```diff
@@ -343,22 +343,15 @@ describe('PATCH/api/articles/:article_id', () => {
...
-          expect(obj).toHaveProperty("title", expect.any(String))
-          expect(obj).toHaveProperty("topic", expect.any(String))
-          expect(obj).toHaveProperty("created_at", expect.any(String))
-          expect(obj).toHaveProperty("article_img_url", expect.any(String))
-          expect(obj).toHaveProperty("body", expect.any(String))
-          expect(obj).toHaveProperty("votes")
-          expect(obj.votes).toBe(106) 
-      });
+
+        expect(articleObj.article.votes).toBe(106)
```
Comment (Augs0) [inline __tests__/endpoint.test.js]: We seem to have a double line here. For what it's worth, I would keep the check for all the properties, just not in a loop :)

File: models/model.js [hunk 13]
```diff
@@ -96,6 +96,12 @@ const { vote_increment } = newComment;
     
 
   ).then((result) => {
+    if (result.rows.length === 0) {
+    return Promise.reject({
+      status: 404,
+      msg: `No user found for article: ${article_id}`,
```
Comment (Augs0) [inline models/model.js]: Careful, it's not the user we have not found, but the article that wasn't found. We don't expect a user in the request so there would be no way to know if the user exists or not

**Review by Augs0 — APPROVED** [review]
Hey, well done! There's a few things to clean up but the actual functionality is there. Take a few minutes to just read the comments here and edit accordingly then you can merge.

**Pull request merged**